lib/api_entity.rb

```diff
@@ -45,18 +45,21 @@ def url
     delegate :relationships, to: :class
 
     def inspect
-      if @attributes.blank?
-        candidate_variables = instance_variables - %i[@attributes @casted_by]
-
-        candidate_variables.each_with_object({}) do |variable, acc|
-          key = variable.to_s.sub('@', '')
-          acc[key] = public_send(key)
-        end
-      else
-        keys_to_exclude = relationships.to_a + [:casted_by]
-
-        @attributes.except(*keys_to_exclude)
-      end
+      inspection = if defined?(@attributes) && @attributes.any?
```

Comment:
Reads quite a lot more complex than before (lots happening in a condensed space)

Reply:
Yeh, I was trying to avoid adding a nest of methods just for inspect but I'll split it out into private methods